src/node/snapshot_serdes.h

```diff
@@ -71,62 +50,56 @@ namespace ccf
     auto tx_hdr = serialized::peek<kv::SerialisedEntryHeader>(data, size);
     auto store_snapshot_size = sizeof(kv::SerialisedEntryHeader) + tx_hdr.size;
 
-    // Snapshots without a snapshot evidence seqno specified by the host should
-    // be self-verifiable with embedded receipt
-    if (!evidence_seqno.has_value())
+    auto receipt_data = data + store_snapshot_size;
+    auto receipt_size = size - store_snapshot_size;
+
```

Comment:
An early check here, with an error message if the receipt_size is empty to point out that this is unsupported snapshot may be good.